src/blas/backends/cublas/cublas_level1.cpp

```diff
@@ -0,0 +1,604 @@
+        });
+    });
+}
+
+#define ROT_LAUNCHER(TYPE1, TYPE2, TYPE3, CUBLAS_ROUTINE)                                          \
+    void rot(cl::sycl::queue &queue, int64_t n, cl::sycl::buffer<TYPE1, 1> &x, const int64_t incx, \
+             cl::sycl::buffer<TYPE1, 1> &y, int64_t incy, TYPE2 c, TYPE3 s) {                      \
+        rot(CUBLAS_ROUTINE, queue, n, x, incx, y, incy, c, s);                                     \
+    }
+
+ROT_LAUNCHER(float, float, float, cublasSrot)
+ROT_LAUNCHER(double, double, double, cublasDrot)
+ROT_LAUNCHER(std::complex<float>, float, float, cublasCsrot)
+ROT_LAUNCHER(std::complex<double>, double, double, cublasZdrot)
+#undef ROT_LAUNCHER
+
+void sdsdot(cl::sycl::queue &queue, int64_t n, float sb, cl::sycl::buffer<float, 1> &x,
+            int64_t incx, cl::sycl::buffer<float, 1> &y, int64_t incy,
+            cl::sycl::buffer<float, 1> &result) {
+    overflow_check(n, incx, incy);
+    // cuda does not support sdot so we need to mimic sdot
+    queue.submit([&](cl::sycl::handler &cgh) {
+        auto x_acc   = x.get_access<cl::sycl::access::mode::read>(cgh);
+        auto y_acc   = y.get_access<cl::sycl::access::mode::read>(cgh);
+        auto res_acc = result.get_access<cl::sycl::access::mode::write>(cgh);
+        cgh.interop_task([=](cl::sycl::interop_handler ih) {
+            auto sc     = CublasScopedContextHandler(queue);
+            auto handle = sc.get_handle(queue);
+            // By default the pointer mode is the CUBLAS_POINTER_MODE_HOST
+            // when the data is on buffer, it must be set to
+            // CUBLAS_POINTER_MODE_DEVICE mode otherwise it causes the segmentation
+            // fault. When it is said to device it is users responsibility to
+            // synchronise as the function is completely asynchronous.
+            cublasSetPointerMode(handle, CUBLAS_POINTER_MODE_DEVICE);
+            auto x_   = sc.get_mem<float *>(ih, x_acc);
+            auto y_   = sc.get_mem<float *>(ih, y_acc);
+            auto res_ = sc.get_mem<float *>(ih, res_acc);
+            cublasStatus_t err;
+            CUBLAS_ERROR_FUNC(cublasSdot, err, handle, n, x_, incx, y_, incy, res_);
+        });
+    });
+    // Since SB is a host pointer we need to bring the result back to the host and
+    // add sb to it.
+    result.get_access<cl::sycl::access::mode::read_write>()[0] += sb;
+}
+
+void dot(cl::sycl::queue &queue, int64_t n, cl::sycl::buffer<float, 1> &x, int64_t incx,
+         cl::sycl::buffer<float, 1> &y, int64_t incy, cl::sycl::buffer<double, 1> &result) {
+    overflow_check(n, incx, incy);
+    // cuda does not support sdot so we need to mimic sdot
+    // converting float* to double * is very constly as sycl reinterpret does not
+    // support conversion from two types which is not the same size. So in order,
+    // to avoid loosing performance we are converting the result to be the float
+    // FIXME:: this change may cause failiour as the result precision reduces.
```

Comment:
FIXME? Is this supposed to be fixed already?

Reply:
Yes, the comment was outdated, I have updated the comments.